src/components/Switch.js

```diff
@@ -16,59 +16,39 @@ const propTypes = {
 };
 
-const PressableWithFeedback = Pressables.PressableWithFeedback;
-class Switch extends Component {
-    constructor(props) {
-        super(props);
-        this.offPosition = 0;
-        this.onPosition = 20;
-        this.offsetX = new Animated.Value(props.isOn ? this.onPosition : this.offPosition);
-
-        this.toggleSwitch = this.toggleSwitch.bind(this);
-        this.toggleAction = this.toggleAction.bind(this);
-    }
-
-    componentDidUpdate(prevProps) {
-        if (prevProps.isOn === this.props.isOn) {
-            return;
-        }
+const OFFSET_X = {
+    OFF: 0,
+    ON: 20,
+};
 
-        this.toggleSwitch();
-    }
+function Switch(props) {
+    const offsetX = useRef(new Animated.Value(props.isOn ? OFFSET_X.ON : OFFSET_X.OFF)).current;
 
-    // animates the switch to the on or off position
-    toggleSwitch() {
-        Animated.timing(this.offsetX, {
-            toValue: this.props.isOn ? this.onPosition : this.offPosition,
+    useEffect(() => {
+        Animated.timing(offsetX, {
+            toValue: props.isOn ? OFFSET_X.ON : OFFSET_X.OFF,
             duration: 300,
             useNativeDriver: true,
         }).start();
-    }
-
-    // executes the callback passed in as a prop
-    toggleAction() {
-        this.props.onToggle(!this.props.isOn);
-    }
-
-    render() {
-        const switchTransform = {transform: [{translateX: this.offsetX}]};
-        return (
-            <PressableWithFeedback
-                style={[styles.switchTrack, !this.props.isOn && styles.switchInactive]}
-                onPress={this.toggleAction}
-                onLongPress={this.toggleAction}
-                accessibilityRole={CONST.ACCESSIBILITY_ROLE.SWITCH}
-                accessibilityState={{checked: this.props.isOn}}
-                aria-checked={this.props.isOn}
-                accessibilityLabel={this.props.accessibilityLabel}
-                // disable hover dim for switch
-                hoverDimmingValue={1}
-                pressDimmingValue={0.8}
-            >
-                <Animated.View style={[styles.switchThumb, switchTransform]} />
-            </PressableWithFeedback>
-        );
-    }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [props.isOn]);
+
+    return (
+        <PressableWithFeedback
+            style={[styles.switchTrack, !props.isOn && styles.switchInactive]}
+            onPress={() => props.onToggle(!props.isOn)}
+            onLongPress={() => props.onToggle(!props.isOn)}
+            accessibilityRole={CONST.ACCESSIBILITY_ROLE.SWITCH}
+            accessibilityState={{checked: props.isOn}}
+            aria-checked={props.isOn}
+            accessibilityLabel={props.accessibilityLabel}
+            // disable hover dim for switch
+            hoverDimmingValue={1}
+            pressDimmingValue={0.8}
+        >
+            <Animated.View style={[styles.switchThumb, {transform: [{translateX: offsetX}]}]} />
```

Comment:
No inline styles. Please write a functional style that accepts `offsetX` as a parameter.

Reply:
This is still not done. What I meant is to write a function that returns a style like `textInputLabelTransformation` or `floatingMessageCounterTransformation`.